### PR TITLE
fixes single and all one state lineages

### LINKED ISSUE #288: Throw out only single state lineages
So we don't get pathological cases

### PR DESCRIPTION
**Pull request by shak360**
fixes #288 and implements #289

### CODE REVIEW

File: lineage/LineageTree.py [hunk 1]
```diff
@@ -178,32 +178,27 @@ def __repr__(self):
...
-            s_list = []
-            for state in range(self.num_states):
-                s_list.append("\n \t There are {} cells of state {}".format(self.lineage_stats[state].num_output_lin_cells, state))
-            s2 = seperator.join(s_list)
-            s3 = ".\n This UNcensord tree has {} many cells in total".format(len(self.output_lineage))
-        return s1 + s2 + s3
+        s = ""
+        for cell in self.output_lineage:
+            s += cell.__repr__()
+        return s
```
Comment (shak360) [inline lineage/LineageTree.py:184]: new print function for lineages

File: lineage/LineageTree.py [hunk 2]
```diff
@@ -178,32 +178,27 @@ def __repr__(self):
...
 
     def __str__(self):
         """
         This function is used to get string representation of an object,
         used for showing the results to the user.
         """
         return self.__repr__()
+
```
Comment (shak360) [inline lineage/LineageTree.py:192]: checks or metrics that help define "good" lineages

File: lineage/figures/figureCommon.py [hunk 3]
```diff
@@ -15,16 +15,18 @@
 pi = np.array([0.5, 0.5], dtype="float")
 
 # T: transition probability matrix
-T = np.array([[0.95, 0.05], [0.05, 0.95]], dtype="float")
+T = np.array([[0.9, 0.1], [0.1, 0.9]], dtype="float")
 
 # bern, gamma_a, gamma_scale
-state0 = StateDistribution(0.99, 20, 5)
-state1 = StateDistribution(0.75, 10, 1)
+state0 = StateDistribution(0.99, 7, 7)
```
Comment (shak360) [inline lineage/figures/figureCommon.py:21]: new fixed gamma parameter values,
shape value is constant,
scale value is varying

File: lineage/figures/figureCommon.py [hunk 4]
```diff
@@ -34,6 +36,13 @@
 num_data_points = 50
 
 
+def lineage_good_to_analyze(tmp_lineage):
```
Comment (shak360) [inline lineage/figures/figureCommon.py:39]: checker function that checks whether or not a lineage is good for analysing

Comment (aarmey) [reply lineage/figures/figureCommon.py:39]: Something like your comment here should be the docstring.

File: lineage/figures/figure9.py [hunk 5]
```diff
@@ -49,15 +50,20 @@ def accuracy():
...
     list_of_populations = []
     list_of_fpi = []
     list_of_fT = []
     list_of_fE = []
     for E in list_of_Es:
         population = []
 
-        population.append(LineageTree(pi, T, E, max_desired_num_cells, censor_condition=3, desired_experiment_time=max_experiment_time))
+        good2go = false
+        while not good2go:
```
Comment (aarmey) [inline lineage/figures/figure9.py:62]: See my simplification here, and change the other instances.

File: lineage/LineageTree.py [hunk 6]
```diff
@@ -178,32 +178,27 @@ def __repr__(self):
...
+    
+    def __len__(self):
+        return len(self.output_lineage)
+    
+    def is_heterogeneous(self):
+        true_states = [cell.state for cell in self.output_lineage]
+        if len(set(true_states)) > 1:
+            return True
+        else:
+            return False
```
Comment (aarmey) [inline lineage/LineageTree.py:201]: Don't need the else, because you'll only get here if you skip the if statement.

**Pull request merged**